src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -499,4 +512,27 @@ public void updateUsedClassWithQualifiedClassName(String qualifiedClassName) {
       updateUsedClassWithQualifiedClassName(potentialOuterClass);
     }
   }
+
+  /**
+   * Given the resolved type of a used member, this method will update the list of used classes
+   * based on the type arguments of the input type accordingly.
+   *
+   * <p>We should always call this method whenever we call {@link
+   * TargetMethodFinderVisitor#updateUsedClassWithQualifiedClassName(String)}.
```

Comment:
This documentation is inconsistent with what you said in your [PR comment](https://github.com/kelloggm/specimin/pull/92/files#r1469646362), where you said:
> It's not always the case that these two methods appear together. These two methods only appear together when we are updating the list of used classes based on the types of some used members.

However, this documentation says "we should always call this method whenever we call...", which suggests that the two methods should always appear together.

From your earlier comment, I had understood the requirement to be that a call to this method (`updateUsedClassWithPossibleTypeArgs`) must always _follow_ a call to the other method (`updateUsedClassWithQualifiedClassName`). However, not all calls to that method need to have a call to this method after them. Is that correct? If so, please correct this documentation. Otherwise, please correct my understanding.

Reply:
The documentation on `updateUsedClassWithPossibleTypeArgs` remains inconsistent with what you've said in this comment: the documentation says:
> We should **always** call this method **whenever** we call {@link
>  TargetMethodFinderVisitor#updateUsedClassWithQualifiedClassName(String)}.

(emphasis mine.) This differs from what you're saying in these PRs comments (and in the documentation of the other method), which suggest that calling `updateUsedClassWithPossibleTypeArgs` should only be called when `updateUsedClassWithQualifiedClassName` is called on a used member. The documentation of one of the two methods _must_ change, because currently they contradict each other.

Stepping back, might it be possible to encode this in the program itself rather than require the user to understand the relationship of these two methods? In particular, could `updateUsedClassWithQualifiedClassName` check whether its argument is a member of the `usedMembers` field, and if so, call `updateUsedClassWithPossibleTypeArgs` automatically? If we did that, then you could 1) remove all the new calls to `updateUsedClassWithPossibleTypeArgs` in this PR, 2) remove the complexity in this documentation, and 3) prevent any missed calls in the future automatically.

Reply:
I think my PR comments have been unclear.

Given a used member, we need to add to the list of used classes:
1: The class containing the declaration of that member.
2: The class corresponding to the type of that member.

If we are doing the first task, then we only need to call `updateUsedClassWithQualifiedClassName`. If we are doing the second task, then we need to call both `updateUsedClassWithQualifiedClassName` and `updateUsedClassWithPossibleTypeArgs`. So `updateUsedClassWithPossibleTypeArgs` will always be called together with `updateUsedClassWithQualifiedClassName` (the second task), while `updateUsedClassWithQualifiedClassName` might be called independently (in case of the first task).

That's why in the documentation of `updateUsedClassWithQualifiedClassName`, I said that we also need to call `updateUsedClassWithPossibleTypeArgs` when we are updating the type of a used member (which is the second task).

> Stepping back, might it be possible to encode this in the program itself rather than require the user to understand the relationship of these two methods? 

I think it will be quite challenging to write codes to differentiate between the class containing a declaration of a member and the class corresponding to the type of that member.

Reply:
> If we are doing the second task, then we need to call both updateUsedClassWithQualifiedClassName and updateUsedClassWithPossibleTypeArgs

In that case, we should have a method that calls both whose documentation specifically calls out this case. What I want us to avoid is a maintainer needing to call two methods to cover one case but only call one of the two to cover the other - it is always better to have _one_ method to call that covers each situation.